src/components/CIPipelineN/VariableContainer.tsx

```diff
@@ -0,0 +1,83 @@
+import Tippy from '@tippyjs/react'
+
+export function VariableContainer({ type }: { type: PluginVariableType }) {
+    const [collapsedSection, setCollapsedSection] = useState<boolean>(true)
+    const {
+        formData,
+        selectedTaskIndex,
+        activeStageName,
+        formDataErrorObj,
+    }: {
+        formData: FormType
+        selectedTaskIndex: number
+        activeStageName: string
+        formDataErrorObj: FormErrorObjectType
+    } = useContext(ciPipelineContext)
+    return (
+        <div>
+            <div
+                className="mb-10 flexbox pointer"
+                onClick={(event) => {
+                    setCollapsedSection(!collapsedSection)
+                }}
+            >
+                <span className="fw-6 fs-13 cn-9">{type} variables</span>
+
+                <Dropdown className="mt-10" style={{ transform: collapsedSection ? 'rotate(180deg)' : 'rotate(0)' }} />
+            </div>
+            {!collapsedSection && (
+                <div className="variable-container">
+                    <div className="fs-12 fw-6 text-uppercase">Variable</div>
+                    <div className="fs-12 fw-6 text-uppercase">Format</div>
+                    <div className="fs-12 fw-6 text-uppercase">
+                        {type === PluginVariableType.INPUT ? 'Value' : 'Description'}
+                    </div>
+                    {formData[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail[
+                        type === PluginVariableType.OUTPUT ? 'outputVariables' : 'inputVariables'
+                    ]?.map((variable, index) => (
+                        <>
+                            {type === PluginVariableType.INPUT && variable.description ? (
+                                <Tippy className="default-tt" arrow={false} content={variable.description}>
+                                    <div className="fs-13 fw-4 lh-28">
+                                        <span className="text-underline-dashed">{variable.name}</span>
+                                    </div>
+                                </Tippy>
+                            ) : (
+                                <div className="fs-13 fw-4 lh-28">{variable.name}</div>
+                            )}
+
+                            <div className="fs-13 fw-4 lh-28">{variable.format}</div>
+                            {type === PluginVariableType.INPUT ? (
+                                <div className="fs-14">
+                                    <CustomInputVariableSelect selectedVariableIndex={index} />
+                                    {formDataErrorObj[activeStageName].steps[selectedTaskIndex]?.pluginRefStepDetail
+                                        .inputVariables[index] &&
+                                        !formDataErrorObj[activeStageName].steps[selectedTaskIndex]?.pluginRefStepDetail
+                                            .inputVariables[index].isValid && (
```

Comment:
We can use optional - `!formDataErrorObj[activeStageName].steps[selectedTaskIndex]?.pluginRefStepDetail
                                            .inputVariables[index]?.isValid`

Reply:
This will not solve the purpose error should only be shown when we have that parent object and it is not valid. If we use optional it will be showing when we dont have that parent as well

Reply:
Ok got the idea here.
Can we move `formData[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail` to some variable at the top & use it as it's being ref'ed at multiple places